Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Search.hs

```diff
@@ -136,3 +144,35 @@ buildMultipleRoutesTag :: Spec.SearchReqMessage -> Maybe [Maps.RouteInfo]
 buildMultipleRoutesTag req = do
   let tagGroups = req.searchReqMessageIntent >>= (.intentFulfillment) >>= (.fulfillmentTags)
   decode . encodeUtf8 =<< Utils.getTagV2 Tag.ROUTE_INFO Tag.MULTIPLE_ROUTES tagGroups
+
+validateSubscriber :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Text -> Id DM.Merchant -> Id MOC.MerchantOperatingCity -> m ()
+validateSubscriber subscriberId merchantId merchantOperatingCityId = do
+  totalSubIds <- QWhiteList.countTotalSubscribers
```

Comment:
add `merchantId` and `merchantOperatingCityId` to both `countTotalSubscribers` and `checkBlacklisted`